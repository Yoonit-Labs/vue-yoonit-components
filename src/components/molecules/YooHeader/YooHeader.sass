@import '~@yoo/style/web/themes/key'
@import '~@yoo/style/web/molecules/YooHeader'