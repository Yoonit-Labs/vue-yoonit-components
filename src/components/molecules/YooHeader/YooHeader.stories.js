import yooHeader from './YooHeader.vue'
import yooButton from '@/components/atoms/Button/Button.vue'

export default {
  title: 'Yoonit/Header',
  component: yooHeader
}

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { yooHeader, yooButton },
  template:
    `<yoo-header v-bind="$props">
      <template v-slot:header-left>
        <yoo-button text="" icon="chevron-left" variation="clear"></yoo-button>
      </template>

      <template v-slot:header-center>
        <img src="https://user-images.githubusercontent.com/7132697/104505155-408e0e00-55c2-11eb-9571-65e1377d7560.png" style="display:flex; margin: 0 auto;" width="40%" alt="" />
      </template>

      <template v-slot:header-right>
        <yoo-button text="" icon="user" variation="clear"></yoo-button>
      </template>
    </yoo-header>`
})

export const Custom = Template.bind({})
Custom.slots = { text: 'Yoo Warning test component' }
