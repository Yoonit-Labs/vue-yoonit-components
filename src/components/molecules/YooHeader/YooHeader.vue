<template lang="pug">
.yoo-header
  YooFlexLayout(
    justifyContent="space-between"
    alignItems="center"
    alignContent="center"
  )
    Div
      slot(
        name="header-left"
      )
    Div
      slot(
        name="header-center"
      )
    Div
      slot(
        name="header-right"
      )
</template>

<script>

import YooFlexLayout from '@/components/quarks/YooFlexLayout/YooFlexLayout.vue'

export default {
  name: 'YooHeader',
  data: () => ({}),
  created () {},
  mounted () {},
  updated () {},
  beforeDestroy () {},
  destroyed () {},
  components: {
    YooFlexLayout
  },
  computed: {},
  methods: {},
  filters: {},
  watch: {}
}
</script>

<style src="./YooHeader.sass" lang="sass" scoped></style>
