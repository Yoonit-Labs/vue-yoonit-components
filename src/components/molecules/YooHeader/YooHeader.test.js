import { shallowMount } from '@vue/test-utils'
import YooHeader from '@/components/molecules/YooHeader/YooHeader.vue'

const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return shallowMount(YooHeader, {
    slots: { default: SlotText }
  })
}

describe('YooHeader Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    expect(wrapper.find('.yoo-header').exists()).toBe(true)
  })
})
