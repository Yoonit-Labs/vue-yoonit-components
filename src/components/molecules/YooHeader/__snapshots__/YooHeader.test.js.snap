// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooHeader Component Matches Snapshot 1`] = `
<div class="yoo-header">
  <yooflexlayout-stub flexdirection="row" flexwrap="nowrap" justifycontent="space-between" alignitems="center" aligncontent="center" gap="0">
    <div>
      <p>Slot left</p>
    </div>
    <div>
      <p>Slot center</p>
    </div>
    <div>
      <p>Slot right</p>
    </div>
  </yooflexlayout-stub>
</div>
`;
